accesscontroltool-bundle/src/main/java/biz/netcentric/cq/tools/actool/authorizableutils/impl/ExternalGroupCreatorServiceImpl.java

```diff
@@ -0,0 +1,128 @@
+
+    public Authorizable createGroupWithExternalId(
+            final UserManager userManager,
+            final AuthorizableConfigBean authorizableConfigBean,
+            AcInstallationHistoryPojo status,
+            AuthorizableInstallationHistory authorizableInstallationHistory,
+            ValueFactory vf,
+            Map<String, Set<AuthorizableConfigBean>> principalMapFromConfig, Session session)
+            throws AuthorizableExistsException, RepositoryException,
+            AuthorizableCreatorException {
+
+        if (StringUtils.isBlank(authorizableConfigBean.getExternalId())) {
+            throw new IllegalStateException("externalId must not be emtpy for " + authorizableConfigBean);
+        }
+
+        ExternalGroup externalGroup = new PrecreatedExternalGroup(authorizableConfigBean);
+
+        ExternalGroupPrecreatorSyncContext externalGroupPrecreatorSyncContext = new ExternalGroupPrecreatorSyncContext(userManager, vf);
+        Group group = externalGroupPrecreatorSyncContext.createExternalGroup(externalGroup);
+
+        return group;
+    }
+
+    // simple workaround to make protected method available here
+    private final class ExternalGroupPrecreatorSyncContext extends DefaultSyncContext {
+
+        private ExternalGroupPrecreatorSyncContext(UserManager userManager, ValueFactory valueFactory) {
+            super(new DefaultSyncConfig(), null, userManager, valueFactory);
+        }
+
+        private Group createExternalGroup(ExternalGroup eg) throws RepositoryException {
+            config.group().setPathPrefix("");
+            return createGroup(eg);
+        }
+    }
+
+    // mapping AuthorizableConfigBean -> ExternalGroup
+    private final class PrecreatedExternalGroup implements ExternalGroup {
+        private final AuthorizableConfigBean authorizableConfigBean;
+
+        private PrecreatedExternalGroup(AuthorizableConfigBean authorizableConfigBean) {
+            this.authorizableConfigBean = authorizableConfigBean;
+        }
+
+        @Override
+        public String getId() {
+            return authorizableConfigBean.getPrincipalID();
+        }
+
+        @Override
+        public String getPrincipalName() {
+            String principalName = ExternalIdentityRef.fromString(authorizableConfigBean.getExternalId()).getId();
+            return principalName;
+        }
+
+        @Override
+        public String getIntermediatePath() {
+            String rawIntermediatePath = authorizableConfigBean.getPath();
+            String intermediatePath = StringUtils.removeStart(rawIntermediatePath, Constants.GROUPS_ROOT + "/");
+            return intermediatePath;
+        }
+
+        @Override
+        public ExternalIdentityRef getExternalId() {
+            return ExternalIdentityRef.fromString(authorizableConfigBean.getExternalId());
+        }
+
+        @Override
+        public Map<String, ?> getProperties() {
+            return Collections.<String, Object> emptyMap();
```

Comment:
isn't it safer to just throw http://docs.oracle.com/javase/8/docs/api/?java/lang/UnsupportedOperationException.html for all methods which are not supposed to be used?